Adds support for vuln scanning binary and go content types with grype

zhill opened the pull request:
Adds cpe generation where needed in the sbom generation step prior to
handoff to Grype.

Also updates to latest Grype release that supports go-module vuln scans
and a default matcher that will handle 'binary' types.

Signed-off-by: Zach Hill <zach@anchore.com>

nightfurys commented on tests/unit/anchore_engine/services/policy_engine/engine/vulns/test_mappers.py:
```diff
@@ -152,427 +152,508 @@ def test_image_content_to_grype_metadata(self, input_metadata, expected_output):
-                        "sourceRpm": None,
-                        "epoch": 1,
-                    },
+    pass
```
Any reason why this was added and the tests are out of the class?

zhill replied:
hmm, this doesn't look right. I'll fix this

zhill commented:
The cpe generation code is lifted from the policy engine loader. I moved it and cleaned it up a bit to ensure no weird import dependencies and since the call parameters were different. The only generator with any real complexity is the java generator which I only very mildly refactored (moved the tokenizer phase to its own function).

nightfurys reviewed (approved):
lgtm
